### PR TITLE
GEOMESA-1107 Allow AWS S3 via s3a

### PR DESCRIPTION
**Pull request by vpipkt**
Stacks on GEOMESA-1106 (PR #813 ) 
Together enables both s3a and s3n ingest.
Updated the tools readme

### CODE REVIEW

**Comment by jahhulbert-ccri** [comment]
@vpipkt are all the classpaths the same for hadoop 2.7? I'm gonna try that out with cloud local and s3a to see how it goes

File: geomesa-tools/README.md [hunk 1]
```diff
@@ -647,25 +647,63 @@ For more documentation on converter configuration, check out the [geomesa-conver
...
-the Hadoop configuration used with GeoMesa Tools. Edit the core-site.xml file in your Hadoop installation. These
-instructions apply to Hadoop 2.5.0 and higher. Note that you must have the environmental variable
-``HADOOP_MAPRED_HOME`` set properly in your environment.
+Hadoop ships with an implementation of a S3 filesystems that can be enabled in the Hadoop configuration used with GeoMesa Tools. GeoMesa Tools can perform ingest using both the second-generation (`s3n`) and third-generation (`s3a`) filesystems. Edit the `$HADOOP_CONF_DIR/core-site.xml` file in your Hadoop installation, as shown below. These instructions apply to Hadoop 2.5.0 and higher. Note that you must have the environment variable ``HADOOP_MAPRED_HOME`` set properly in your environment. Some configurations can substitute ``HADOOP_PREFIX`` in the classpath values below.
+
+Warning: AWS credentials are valueable. They pay for services and control read and write protection for data. If you are running GeoMesa on AWS EC2 instances, it is recommended to use s3a. With s3a, you can omit the Access Key Id and Secret Access keys from `core-site.xml` and rely on IAM roles. 
+
+##### s3a
```
Comment (jahhulbert-ccri) [inline geomesa-tools/README.md]: @vpipkt can you also make the doc changes here: https://github.com/locationtech/geomesa/blob/master/docs/user/commandline_tools.rst

**Comment by vpipkt** [comment]
@jahhulbert-ccri Good question about 2.7 -- I'm not sure. Thanks for help with  testing!

**Comment by vpipkt** [comment]
@mdzimmerman  Can you take a look at the doc updates?  my first time working with rst format

**Comment by jahhulbert-ccri** [comment]
Verified ingest worked with hadoop 2.6.4 and accumulo 1.6.5 pending doc analysis by @mdzimmerman i'll merge.

File: docs/user/commandline_tools.rst [hunk 2]
```diff
@@ -299,41 +299,75 @@ Shape files may also be ingested::
...
-        </property>
+        <name>mapreduce.application.classpath</name>
+        <value>$HADOOP_MAPRED_HOME/share/hadoop/mapreduce/*:$HADOOP_MAPRED_HOME/share/hadoop/mapreduce/lib/*:$HADOOP_MAPRED_HOME/share/hadoop/tools/lib/*</value>
+        <description>The classpath specifically for mapreduce jobs. This override is neeeded so that s3 URLs work on hadoop 2.6.0+</description>
+    </property>
 
+    <!-- OMIT these keys if running on AWS EC2; use IAM roles instead -->
+    <!-- These are Hadoop 2.7 style properties -->
+    <property>
+        <name>fs.s3a.awsAccessKeyId</name>
```
Comment (jahhulbert-ccri) [inline docs/user/commandline_tools.rst]: @vpipkt I wasn't able to get these working in 2.7 I had to use the 2.6 style properties...I think that is what the documentation for 2.7 says as well:
https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/index.md#authentication-properties-2

Comment (vpipkt) [inline docs/user/commandline_tools.rst]: There are contradictory documents... I will dig a little deeper and test it out.   https://hadoop.apache.org/docs/r2.7.1/hadoop-project-dist/hadoop-common/core-default.xml

**Comment by vpipkt** [comment]
Indeed the readme was correct and the docs was incorrect...

**Comment by jahhulbert-ccri** [comment]
mergin

**Comment by jahhulbert-ccri** [comment]
9feb4b357dd64d937a7f8e59253afc64d8fb5cf2

**Pull request closed without merging**